apps/arweave/src/ar_block_pre_validator.erl

```diff
@@ -243,21 +239,77 @@ may_be_pre_validate_second_chunk_hash(#block{ recall_byte2 = undefined } = B, Pr
 			invalid;
 		true ->
 			%% The block is not supposed to have the second chunk.
-			pre_validate_indep_hash(B, PrevB, Peer)
+			may_be_pre_validate_first_unpacked_chunk_hash(B, PrevB, Peer)
 	end;
-may_be_pre_validate_second_chunk_hash(#block{ poa2 = #poa{ chunk = <<>> } } = B, PrevB,
-		Peer) ->
-	%% The second chunk has not been sent along (we should have informed the sender we've got
-	%% the chunk so we can read it locally) => validate the second chunk hash later.
-	pre_validate_indep_hash(B, PrevB, Peer);
 may_be_pre_validate_second_chunk_hash(B, PrevB, Peer) ->
 	case crypto:hash(sha256, (B#block.poa2)#poa.chunk) == B#block.chunk2_hash of
 		false ->
 			post_block_reject_warn(B, check_second_chunk, Peer),
 			ar_events:send(block, {rejected, invalid_second_chunk, B#block.indep_hash, Peer}),
 			invalid;
+		true ->
+			may_be_pre_validate_first_unpacked_chunk_hash(B, PrevB, Peer)
+	end.
+
+may_be_pre_validate_first_unpacked_chunk_hash(
+		#block{ packing_difficulty = PackingDifficulty } = B, PrevB, Peer)
+			when PackingDifficulty >= 1 ->
+	PoA = B#block.poa,
+	case crypto:hash(sha256, PoA#poa.unpacked_chunk) == B#block.unpacked_chunk_hash
+			%% The unpacked chunk is expected to be 0-padded when smaller than
+			%% ?DATA_CHUNK_SIZE.
+			andalso byte_size(PoA#poa.unpacked_chunk) == ?DATA_CHUNK_SIZE of
+		false ->
+			post_block_reject_warn(B, check_first_unpacked_chunk, Peer),
+			ar_events:send(block, {rejected, invalid_first_unpacked_chunk,
```

Comment:
Do we need to add `invalid_first_unpacked_chunk` to the `ar_peers.erl` `?BLOC_REJECTION_IGNORE` list?

Reply:
Oops, I hadn't refreshed. I see it now!